compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,6 +30,116 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
+typedef enum RegisterStatus {
+   OK_FOR_REUSE,
+   DO_NOT_TOUCH
+} RegisterStatus;
+
+static inline RegisterStatus
+evaluateRegister(TR::Register **childReg, TR::Node *child, TR::Node *parent, TR::CodeGenerator *cg)
+   {
+   int64_t value;
+   RegisterStatus status;
+
+	if (child->getOpCode().isLoadConst() && child->getRegister() == NULL)
+   {
+      value = child->getLongInt();
+      if (constantIsUnsignedImm12(value))
+      {
+         //TODO: Cover special cases for immediate loads
+      }
+      else
+      {
+         *childReg = cg->allocateRegister();
+         loadConstant64(cg, parent, value, *childReg);
+         status = OK_FOR_REUSE;
+      }
+	}
+	else
+   {
+      *childReg = cg->evaluate(child);
+      status = DO_NOT_TOUCH;
+	}
+   return status;
+   }
+
+static inline TR::Register *
+evaluateBooleanTypeInteger(TR::Node *node, TR::InstOpCode::Mnemonic regOp, TR::CodeGenerator *cg, uint8_t sizeOfOperation)
+   {
+   TR::Register *srcReg[3];
+   RegisterStatus srcStatus[3];
+   TR::Node *children[3];
+   TR::Register *trgReg;
+   bool assigned = false;
+
+   if(sizeOfOperation > 0)
+   {
+      children[0] = node->getFirstChild();
+      srcStatus[0] = evaluateRegister(&srcReg[0], children[0], node, cg);
+   }
+   if(sizeOfOperation > 1)
+   {
+      children[1] = node->getSecondChild();
+      srcStatus[1] = evaluateRegister(&srcReg[1], children[1], node, cg);
+   }
+   if(sizeOfOperation > 2)
+   {
+      children[2] = node->getThirdChild();
+      srcStatus[2] = evaluateRegister(&srcReg[2], children[2], node, cg);
+   }
+   
+   for(int i=0; i < sizeOfOperation && !assigned; i++)
+   {
+      if(srcStatus[i] == OK_FOR_REUSE)
+	   {
+	      trgReg = srcReg[i];
+		   assigned = true;
+		   srcStatus[i] = DO_NOT_TOUCH;
```

Comment:
Shouldn't this break from the loop, or do you always want the target to point to the last srcReg found?

Reply:
I think this loop will break when assigned is set to true (as soon as a target is found).

Reply:
Yes, you're right.  Sorry I missed that.